packages/frontend/src/app/article/[slug]/news-reading.tsx

```diff
@@ -0,0 +1,80 @@
+'use client'
+
+import React, { useState, useMemo, useEffect } from 'react'
+import styled from 'styled-components'
+import { Dropdown } from './dropdown'
+
+const Container = styled.div`
+  max-width: 600px;
+  width: calc(280 / 320 * 100%);
+  border: 1px solid #eaeaea;
+  border-radius: 20px;
+  margin-left: auto;
+  margin-right: auto;
+  margin-bottom: 70px;
+`
+
+const Divider = styled.div`
+  border: 1px solid #eaeaea;
+  margin-bottom: 20px;
+`
+
+const IframeContainer = styled.div`
+  padding: 20px;
+`
+
+type NewsReadingProps = {
+  className?: string
+  data: {
+    items: {
+      name: string
+      embedCode: string
+    }[]
+  }
+}
+
+const NewsReading = function ({ className, data }: NewsReadingProps) {
+  const [isMounted, setIsMounted] = useState(false)
+  const { items } = data
+
+  const options = useMemo(
+    () =>
+      items.map((r) => {
+        return {
+          name: r.name,
+          value: r.name,
+        }
+      }),
+    [items]
+  )
+  const [selectedOption, setSelectedOption] = useState(options[0])
+  const selectedReading = items.find((r) => r.name === selectedOption.value)
```

Comment:
same as above, could we add error handling for `options?.[0]` `items?.map` `items?.find`